peer/brontide.go

```diff
@@ -3617,6 +3617,22 @@ func (p *Brontide) StartTime() time.Time {
 // message is received from the remote peer. We'll use this message to advance
 // the chan closer state machine.
 func (p *Brontide) handleCloseMsg(msg *closeMsg) {
+	link := p.fetchLinkFromKeyAndCid(msg.cid)
+	if link == nil {
+		// TODO(proofofkeags): is this ever a valid situation?
```

Comment:
I don't follow, can you elaborate? Currently in the code as written, link can be nil here and we'd still need to process the closing_signed. I think we just need to process the closing_signed even if the link is down

Reply:
I currently am fetching the link because later in this function I need to add link lifecycle hooks (introduced in an earlier commit in this PR). It seems like if the link has already been removed then we wouldn't need to insert those hooks. I will update the PR to reflect this understanding and maybe the code will be clearer than my explanation.